api/graphql/schema.graphql

```diff
@@ -1765,13 +1758,17 @@ type Mutation {
     @create(model: "Pool")
     @guard
     @can(ability: "create", model: "Pool")
+  updatePool(id: ID!, pool: UpdatePoolInput! @spread): Pool
```

Comment:
While we're here, let's deprecate updatePool. Reason: should use updatePoolAdvertisement instead.

Reply:
Updated with [a698557](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5927/commits/a698557b0ed82039f4375607024dfdaa1bc6b7db).